modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/FailHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx.impl;
+
+/**
+ * A transaction fail handler.
+ */
+public interface FailHandler {
+    /**
+     * Check if the provided exception is recoverable.
+     *
+     * @param throwable Exception to test.
+     * @return {@code true} if recoverable, {@code false} otherwise.
+     */
+    boolean checkRecoverable(Throwable throwable);
```

Comment:
Just remove the FailHandler for now. And move isRecoverable to where it's currently needed. BTW static?